main.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+type node struct {
+	val         int
+	left, right *node
+}
+
+func levelOrderTraversal(root *node, sink func(v int)) {
+	for i := 0; i < height(root)+1; i++ {
+		traverse(root, i, sink)
+	}
+}
```

Comment:
The solution is correct, but what is its time complexity? Can you invent a solution with a better time complexity (perhaps using an additional data structure)?